packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,292 @@
+        //     numCreditsMin: 4,
+        //     numCreditsMax: 4,
+        //     id: null,
+        //   });
+        // })
+        SearchAPI.fetchCourse(
+          requirement.subject,
+          requirement.classId.toString()
+        )
+      );
+    }
+
+    Promise.all(promises).then((courses) => {
+      const courseMap: { [id: string]: ScheduleCourse2<null> } = {};
+      if (courses) {
+        for (const course of courses) {
+          if (course) {
+            courseMap[`${course.subject}${course.classId}`] = course;
+          }
+        }
+        setCourseData(courseMap);
+      }
+    });
+  }, [major.requirementSections]);
+
+  return (
+    <Box p="xs 0px" backgroundColor="neutral.main">
+      <Text
+        py="lg"
+        px="sm"
+        fontSize="xl"
+        color="primary.red.main"
+        fontWeight={700}
+      >
+        {major.name}
+      </Text>
+      {courseData &&
+        major.requirementSections.map((section, index) => (
+          <SidebarSection
+            key={section.title}
+            section={section}
+            courseData={courseData}
+            dndIdPrefix={"sidebar-" + index}
+          />
+        ))}
+    </Box>
+  );
+});
+
+// We need to manually set the display name like this because
+// of how we're using memo above.
+Sidebar.displayName = "Sidebar";
+
+export { Sidebar };
+
+const SidebarSection: React.FC<SidebarSectionProps> = ({
+  section,
+  courseData,
+  dndIdPrefix,
+}) => {
+  const [opened, setOpened] = useState(false);
+  return (
+    <Box
+      backgroundColor="neutral.main"
+      borderTop="1px solid white"
+      cursor="pointer"
+      userSelect="none"
+    >
+      <Text
+        onClick={() => {
+          setOpened(!opened);
+        }}
+        color="dark.main"
+        fontWeight="bold"
+        py="md"
+        px="sm"
+      >
+        {section.title}
+      </Text>
+      <Box
+        style={{ display: opened ? "" : "none" }}
+        backgroundColor="neutral.900"
+        padding="10px 20px 15px 10px"
+        cursor="default"
+      >
+        {section.requirements.map((requirement, index) => (
+          <SectionRequirement
+            requirement={requirement}
+            courseData={courseData}
+            dndIdPrefix={dndIdPrefix + "-" + index}
+            key={index}
+          />
+        ))}
+      </Box>
+    </Box>
+  );
+};
+
+const SectionRequirement: React.FC<SidebarRequirementProps> = ({
+  requirement,
+  courseData,
+  dndIdPrefix,
+}) => {
+  const renderRequirement = () => {
+    switch (requirement.type) {
+      case "XOM":
+        return renderXOM(requirement);
+      case "AND":
+        return renderAND(requirement);
+      case "OR":
+        return renderOR(requirement);
+      case "RANGE":
+        return renderRange(requirement);
+      case "COURSE":
+        return renderCourse(requirement);
+      case "SECTION":
+        return renderSection(requirement);
+      default:
+        return <p>Unsupported type</p>;
```

Comment:
I think throwing an error here is better than returning unsupported type. Should be some sort of assertUnreachable